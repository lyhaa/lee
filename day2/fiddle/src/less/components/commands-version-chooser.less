button.version-chooser.bp3-button {
  border-radius: 3px 0 0 3px;
}
