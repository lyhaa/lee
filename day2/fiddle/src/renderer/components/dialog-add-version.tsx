
import { Button, Callout, Dialog, FileInput, InputGroup, Intent } from '@blueprintjs/core';
import { observer } from 'mobx-react';
import * as path from 'path';
import * as React from 'react';
import * as semver from 'semver';

import { NpmVersion } from '../../interfaces';
import { getElectronNameForPlatform } from '../../utils/electron-name';
import { AppState } from '../state';

export interface AddVersionDialogProps {
  appState: AppState;
}

export interface AddVersionDialogState {
  isValidElectron: boolean;
  isValidVersion: boolean;
  file?: File;
  version: string;
}

/**
 * The "add version" dialog allows users to add custom builds of Electron.
 *
 * @class AddVersionDialog
 * @extends {React.Component<AddVersionDialogProps, AddVersionDialogState>}
 */
@observer
export class AddVersionDialog extends React.Component<AddVersionDialogProps, AddVersionDialogState> {
  constructor(props: AddVersionDialogProps) {
    super(props);
    
    this.state = {
      isValidVersion: false,
      isValidElectron: false,
      version: ''
    };

    this.onSubmit = this.onSubmit.bind(this);
    this.onClose = this.onClose.bind(this);
    this.onChangeFile = this.onChangeFile.bind(this);
    this.onChangeVersion = this.onChangeVersion.bind(this);
  }

  /**
   * Handles a change of the file input
   *
   * @param {React.ChangeEvent<HTMLInputElement>} event
   */
  public async onChangeFile(event: React.FormEvent<HTMLInputElement>) {
    const { files } = event.target as any;
    const { binaryManager } = this.props.appState;
    const file = files && files[0] ? files[0] : undefined;

    const isValidElectron = !!(file
      && file.path
      && await binaryManager.getIsDownloaded('custom', file.path));

    this.setState({ file, isValidElectron });
  }

  public onChangeVersion(event: React.ChangeEvent<HTMLInputElement>) {
    const version = event.target.value || '';
    const isValidVersion = !!semver.valid(version);

    this.setState({
      version,
      isValidVersion,
     });
  }

  /**
   * Handles the submission of the dialog
   *
   * @returns {Promise<void>}
   */
  public async onSubmit(): Promise<void> {
    const { file, version } = this.state;

    if (!file) return;

    const name = file.path
      .slice(-20)
      .split(path.sep)
      .slice(1)
      .join(path.sep);

    const toAdd: NpmVersion = {
      localPath: file.path,
      version,
      name
    };

    this.props.appState.addLocalVersion(toAdd);
    this.onClose();
  }

  /**
   * Closes the dialog
   */
  public onClose() {
    this.props.appState.isAddVersionDialogShowing = false;
    this.reset();
  }

  get buttons() {
    const canSubmit = this.state.isValidElectron && this.state.isValidVersion;

    return [
      (
        <Button
          icon='add'
          key='submit'
          disabled={!canSubmit}
          onClick={this.onSubmit}
          text='Add'
        />
      ), (
        <Button
          icon='cross'
          key='cancel'
          onClick={this.onClose}
          text='Cancel'
        />
      )
    ];
  }

  public render() {
    const { isAddVersionDialogShowing } = this.props.appState;
    const inputProps = { webkitdirectory: 'true' };
    const { file } = this.state;

    const text = file && file.path
      ? file.path
      : `Select the folder containing ${getElectronNameForPlatform()}...`;

    return (
      <Dialog
        isOpen={isAddVersionDialogShowing}
        onClose={this.onClose}
        title='Add local Electron build'
        className='dialog-add-version'
      >
        <div className='bp3-dialog-body'>
          <FileInput
            onInputChange={this.onChangeFile}
            id='custom-electron-version'
            inputProps={inputProps as any}
            text={text}
          />
          <br />
          {this.renderPath()}
        </div>
        <div className='bp3-dialog-footer'>
          <div className='bp3-dialog-footer-actions'>
            {this.buttons}
          </div>
        </div>
      </Dialog>
    );
  }

  private renderPath(): JSX.Element | null {
    const { file, isValidElectron } = this.state;

    if (!file || !file.path) return null;

    const info = isValidElectron
      ? `We found an ${getElectronNameForPlatform()} in this folder.`
      : `We did not find a ${getElectronNameForPlatform()} in this folder...`;

    return (
      <Callout>
        {info}
        {this.renderVersionInput()}
      </Callout>
    );
  }

  private renderVersionInput(): JSX.Element | null {
    const { isValidElectron, isValidVersion, version } = this.state;

    if (!isValidElectron) return null;

    return (
      <>
        <p>
          Please specify a version, used for typings and the name.
          Must be <code>semver</code> compliant.
        </p>
        <InputGroup
          intent={isValidVersion ? undefined : Intent.DANGER}
          value={version}
          onChange={this.onChangeVersion}
          placeholder='4.0.0'
        />
      </>
    );
  }

  /**
   * Reset this component's state
   */
  private reset(): void {
    this.setState({
      isValidElectron: false,
      isValidVersion: false,
      version: '',
      file: undefined
    });
  }
}
