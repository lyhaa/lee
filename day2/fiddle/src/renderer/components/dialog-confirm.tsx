
import { Alert, Intent } from '@blueprintjs/core';
import { observer } from 'mobx-react';
import * as React from 'react';

import { AppState } from '../state';

export interface ConfirmDialogProps {
  appState: AppState;
}

export interface ConfirmDialogState {
}

/**
 * The token dialog prompts the user to either continue or cancel the operation.
 *
 * @export
 * @class ConfirmDialog
 * @extends {React.Component<ConfirmDialogProps, ConfirmDialogState>}
 */
@observer
export class ConfirmDialog extends React.Component<ConfirmDialogProps, ConfirmDialogState> {
  constructor(props: ConfirmDialogProps) {
    super(props);

    this.onClose = this.onClose.bind(this);
  }

  public onClose(result: boolean) {
    this.props.appState.confirmationPromptLastResult = result;
    this.props.appState.toggleConfirmationPromptDialog();
  }

  public render() {
    const { isConfirmationPromptShowing, confirmationDialogTexts } = this.props.appState;

    return (
      <Alert
        isOpen={isConfirmationPromptShowing}
        onClose={this.onClose}
        icon='help'
        confirmButtonText={confirmationDialogTexts.ok}
        cancelButtonText={confirmationDialogTexts.cancel}
        intent={Intent.PRIMARY}
      >
        <p>{confirmationDialogTexts.label}</p>
      </Alert>
    );
  }
}
