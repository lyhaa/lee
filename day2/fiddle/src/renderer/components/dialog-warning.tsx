
import { Alert, Intent } from '@blueprintjs/core';
import { observer } from 'mobx-react';
import * as React from 'react';

import { AppState } from '../state';

export interface WarningDialogProps {
  appState: AppState;
}

export interface WarningDialogState {
}

/**
 * The token dialog prompts the user to either continue or cancel the operation.
 *
 * @export
 * @class WarningDialog
 * @extends {React.Component<WarningDialogProps, WarningDialogState>}
 */
@observer
export class WarningDialog extends React.Component<WarningDialogProps, WarningDialogState> {
  constructor(props: WarningDialogProps) {
    super(props);

    this.onClose = this.onClose.bind(this);
  }

  public onClose(result: boolean) {
    this.props.appState.warningDialogLastResult = result;
    this.props.appState.toggleWarningDialog();
  }

  public render() {
    const { isWarningDialogShowing, warningDialogTexts } = this.props.appState;
    return (
      <Alert
        isOpen={isWarningDialogShowing}
        onClose={this.onClose}
        icon='warning-sign'
        confirmButtonText={warningDialogTexts.ok}
        cancelButtonText={warningDialogTexts.cancel}
        intent={Intent.DANGER}
      >
        <p>{warningDialogTexts.label}</p>
      </Alert>
    );
  }
}
