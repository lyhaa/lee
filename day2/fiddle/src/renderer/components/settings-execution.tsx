import { Callout, Checkbox, FormGroup } from '@blueprintjs/core';
import { observer } from 'mobx-react';
import * as React from 'react';

import { AppState } from '../state';

export interface ExecutionSettingsProps {
  appState: AppState;
}

/**
 * Settings content to manage execution-related preferences.
 *
 * @class ExecutionSettings
 * @extends {React.Component<ExecutionSettingsProps, {}>}
 */
@observer
export class ExecutionSettings extends React.Component<ExecutionSettingsProps, {}> {
  constructor(props: ExecutionSettingsProps) {
    super(props);

    this.handleDeleteDataChange = this.handleDeleteDataChange.bind(this);
    this.handleElectronLoggingChange = this.handleElectronLoggingChange.bind(this);
  }

  /**
   * Handles a change on whether or not the user data dir should be deleted
   * after a run.
   *
   * @param {React.ChangeEvent<HTMLInputElement>} event
   */
  public handleDeleteDataChange(
    event: React.FormEvent<HTMLInputElement>
  ) {
    const { checked } = event.currentTarget;
    this.props.appState.isKeepingUserDataDirs = checked;
  }

  /**
   * Handles a change on whether or not electron should log more things
   *
   * @param {React.ChangeEvent<HTMLInputElement>} event
   */
  public handleElectronLoggingChange(
    event: React.FormEvent<HTMLInputElement>
  ) {
    const { checked } = event.currentTarget;
    this.props.appState.isEnablingElectronLogging = checked;
  }

  public render() {
    const { isKeepingUserDataDirs, isEnablingElectronLogging } = this.props.appState;

    const deleteUserDirLabel = `
      Whenever Electron runs, it creates a user data directory for cookies, the cache,
      and various other things that it needs to keep around. Since fiddles are usually
      just run once, we delete this directory after your fiddle exits. Enable this
      setting to keep the user data directories around.
    `.trim();
    const electronLoggingLabel = `
      There are some flags that Electron uses to log extra information both internally
      and through Chromium.  Enable this option to make Fiddle produce those logs.`.trim();

    return (
      <div>
        <h2>Execution</h2>
        <Callout>
          These advanced settings control how Electron Fiddle executes your fiddles.
        </Callout>
        <br />
        <Callout>
          <FormGroup label={deleteUserDirLabel}>
            <Checkbox
              checked={isKeepingUserDataDirs}
              label='Do not delete user data directories.'
              onChange={this.handleDeleteDataChange}
            />
          </FormGroup>
        </Callout>
        <br />
        <Callout>
          <FormGroup label={electronLoggingLabel}>
            <Checkbox
              checked={isEnablingElectronLogging}
              label='Enable advanced Electron logging.'
              onChange={this.handleElectronLoggingChange}
            />
          </FormGroup>
        </Callout>
      </div>
    );
  }
}
