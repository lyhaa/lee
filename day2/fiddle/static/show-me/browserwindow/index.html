<!DOCTYPE html>
<html lang="en">
<head>
  <title>BrowserWindow</title>
</head>
<body>
  <h1>Hello from your BrowserWindow!</h1>
  <button id="close-btn">Close this window</button>
  <script>
    const button = document.querySelector('#close-btn')
    button.addEventListener('click', () => window.close())
  </script>
</body>
</html>
