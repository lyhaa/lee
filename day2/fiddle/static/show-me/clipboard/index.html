<!DOCTYPE html>
<html lang="en">
<head>
  <title>Clipboard</title>
</head>
<body>
  <h1>Clipboard Demo</h1>
  <button id="copy">Copy "Hello from Electron!"</button>
  <button id="paste">Paste into TextArea</button>
  <br />
  <textarea></textarea>
  <script>
    require('./renderer.js')
  </script>
</body>
</html>