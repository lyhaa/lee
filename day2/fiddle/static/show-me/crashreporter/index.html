<!DOCTYPE html>
<html lang="en">
<head>
  <title>CrashReporter</title>
</head>
<body>
  <h1>CrashReporter Demo</h1>
  <script>
    require('./renderer.js')
  </script>
</body>
</html>