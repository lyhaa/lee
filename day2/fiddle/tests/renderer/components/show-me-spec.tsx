import { shallow } from 'enzyme';
import * as React from 'react';

import { ShowMe } from '../../../src/renderer/components/show-me';

describe('ShowMe component', () => {
  let mockState: any = {};

  beforeAll(() => {
    mockState = {};
  });

  it('renders', () => {
    const wrapper = shallow(<ShowMe appState={mockState} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a template spec', () => {
    mockState.templateName = 'app';
    const wrapper = shallow(<ShowMe appState={mockState} />);
    expect(wrapper).toMatchSnapshot();
  });
});
