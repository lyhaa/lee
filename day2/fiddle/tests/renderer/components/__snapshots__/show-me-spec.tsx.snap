// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowMe component renders 1`] = `
<div
  className="show-me-panel"
>
  <ShowMeDefault
    appState={Object {}}
  />
</div>
`;

exports[`ShowMe component renders a template spec 1`] = `
<div
  className="show-me-panel"
>
  <ShowMeDefault
    appState={
      Object {
        "templateName": "app",
      }
    }
  />
</div>
`;
